bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/model/MillheatModel.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.millheat.internal.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link MillheatModel} represents the home structure as designed by the user in the Millheat app.
+ *
+ * @author Arne Seime - Initial contribution
+ */
+public class MillheatModel {
+    private long lastUpdated;
+    private List<Home> homes = new ArrayList<>();
+
+    public MillheatModel(long lastUpdated) {
+        this.lastUpdated = lastUpdated;
+    }
+
+    public void addHome(Home home) {
+        homes.add(home);
+    }
+
+    public List<Home> getHomes() {
+        return homes;
+    }
+
+    public long getLastUpdated() {
+        return lastUpdated;
+    }
+
+    public Optional<Heater> findHeaterById(@NonNull String id) {
+        return Stream
```

Comment:
Maybe it might make sense to extra the logic get-stream-of-all-heaters to a separate method instead of repeating it here and on line 69

Reply:
Fixed